fix: avoid crash if ingredients services called without ingredients_lc

stephanegigandet opened the pull request:
fixes #11047

In the future, it would be a good thing to enable the services to add error messages that could be returned to the caller, when there is missing data for instance.

alexgarel commented on lib/ProductOpener/Ingredients.pm:
Didn't you add a test for this scenario ?

stephanegigandet commented:
@alexgarel Thank you for the feedback.

I refactored some things, and then that led to refactoring more things:

- get_or_select_ingredients_lc() : new function called by services that expect ingredients_lc to be set
- select_ingredients_lc() : @language_codes may not be computed is called from a service, so finding existing fields differently
also sets ingredients_text if needed
- APIProductServices: added a way for services to pass back error messages that can be reported in the API response
- added tests for the parse_ingredients_text service

stephanegigandet commented:
/update_tests_results
